pkg/routeagent_driver/handlers/ovn/submarinerNonGWRouteHandler.go

```diff
@@ -0,0 +1,189 @@
+	}
+
+	annotations := node.GetAnnotations()
+
+	transitSwitchIP, ok := annotations["k8s.ovn.org/node-transit-switch-port-ifaddr"]
+	if !ok {
+		logger.Infof("No transit switch IP configured")
+		return nil
+	}
+
+	nonGatewayRouteHandler.transitSwitchIP = transitSwitchIP
+
+	return nil
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) GetName() string {
+	return "submariner-gw-route-handler"
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) GetNetworkPlugins() []string {
+	// TODO enable when we switch to new implementation
+	// return []string{cni.OVNKubernetes}
+	return []string{}
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) LocalEndpointCreated(endpoint *submarinerv1.Endpoint) error {
+	nonGatewayRouteHandler.mutex.Lock()
+	defer nonGatewayRouteHandler.mutex.Unlock()
+
+	nonGatewayRouteHandler.localEndpoint = endpoint
+
+	return nil
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) RemoteEndpointCreated(endpoint *submarinerv1.Endpoint) error {
+	nonGatewayRouteHandler.mutex.Lock()
+	defer nonGatewayRouteHandler.mutex.Unlock()
+
+	if nonGatewayRouteHandler.isGateway && nonGatewayRouteHandler.transitSwitchIP != "" {
+		retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			_, err := nonGatewayRouteHandler.smClient.SubmarinerV1().
+				NonGatewayRoutes(endpoint.Namespace).Create(context.TODO(),
+				nonGatewayRouteHandler.getNonGatewayRoute(endpoint), metav1.CreateOptions{})
+			if !apierrors.IsAlreadyExists(err) {
+				return errors.Wrapf(err, "error processing the remote endpoint create event for %q", endpoint.Name)
+			}
+			return nil
+		})
+
+		return errors.Wrapf(retryErr, "processing the remote endpoint event failed even after retry")
+	}
+
+	return nil
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) RemoteEndpointRemoved(endpoint *submarinerv1.Endpoint) error {
+	nonGatewayRouteHandler.mutex.Lock()
+	defer nonGatewayRouteHandler.mutex.Unlock()
+
+	delete(nonGatewayRouteHandler.remoteEndpoints, endpoint.Name)
+
+	if nonGatewayRouteHandler.isGateway && nonGatewayRouteHandler.transitSwitchIP != "" {
+		if err := nonGatewayRouteHandler.smClient.SubmarinerV1().NonGatewayRoutes(endpoint.Namespace).Delete(context.TODO(),
+			endpoint.Name, metav1.DeleteOptions{}); err != nil {
+			return errors.Wrapf(err, "error deleting nonGatewayRoute %q", endpoint.Name)
+		}
+	}
+
+	return nil
+}
+
+func (nonGatewayRouteHandler *NonGatewayRouteHandler) TransitionToNonGateway() error {
+	nonGatewayRouteHandler.mutex.Lock()
+	defer nonGatewayRouteHandler.mutex.Unlock()
+
+	nonGatewayRouteHandler.isGateway = false
+	if nonGatewayRouteHandler.transitSwitchIP != "" {
+		for _, endpoint := range nonGatewayRouteHandler.remoteEndpoints {
+			if err := nonGatewayRouteHandler.smClient.SubmarinerV1().NonGatewayRoutes(endpoint.Namespace).Delete(context.TODO(),
+				endpoint.Name, metav1.DeleteOptions{}); err != nil {
+				return errors.Wrapf(err, "error deleting nonGatewayRoute %q", endpoint.Name)
```

Comment:
Don't we want to continue to next remote endpoint instead of returning here ?

Reply:
There will be a retry on failure wouldn't that be better?